### PR TITLE
Add TLS fingerprint support

### LINKED ISSUE #51: Update log4j to fix CVE-2021-44228
Cloudwatch adapter uses log4j versions affected by [CVE-2021-44228](https://nvd.nist.gov/vuln/detail/CVE-2021-44228):

```
[INFO]    org.apache.logging.log4j:log4j-core:jar:2.14.1:compile
[INFO]    org.apache.logging.log4j:log4j-slf4j18-impl:jar:2.14.1:compile
[INFO]    com.amazonaws:aws-lambda-java-log4j2:jar:1.2.0:compile
[INFO]    org.apache.logging.log4j:log4j-api:jar:2.14.1:compile
```

Dependency tree

```
[INFO] +- com.amazonaws:aws-lambda-java-log4j2:jar:1.2.0:compile
[INFO] +- org.apache.logging.log4j:log4j-api:jar:2.14.1:compile
[INFO] +- org.apache.logging.log4j:log4j-core:jar:2.14.1:compile
[INFO] +- org.apache.logging.log4j:log4j-slf4j18-impl:jar:2.14.1:compile
```

### LINKED ISSUE #54: Add support for TLS connections
When connecting to Exasol via TLS using self signed certificates it's necessary to specify the certificate fingerprint.

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #54

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -141,19 +142,13 @@
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-core</artifactId>
-            <version>3.11.2</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>com.github.stefanbirkner</groupId>
-            <artifactId>system-lambda</artifactId>
```
Comment (kaklakariada) [inline pom.xml:149]: Not necessary any more after refactoring

File: src/test/java/com/exasol/cloudwatch/LocalstackCloudWatchRaw.java [hunk 2]
```diff
@@ -52,9 +52,8 @@ public LocalstackCloudWatchRaw(final LocalStackContainer localStackContainer) {
 
     private Instant readTimestamp(final String timestampString) {
         final LocalDateTime localDateTime = LocalDateTime.parse(timestampString,
-                DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSS'Z'"));
-        final Instant instant = localDateTime.toInstant(ZoneOffset.UTC);
-        return instant;
+                DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"));
```
Comment (kaklakariada) [inline src/test/java/com/exasol/cloudwatch/LocalstackCloudWatchRaw.java:55]: Parsing timestamp failed with Java 17, `SSS` works with 11 and 17

File: src/test/java/com/exasol/cloudwatch/TestConstants.java [hunk 3]
```diff
@@ -1,6 +1,6 @@
 package com.exasol.cloudwatch;
 
 public class TestConstants {
-    public static final String EXASOL_DOCKER_DB_VERSION = "7.0.5";
-    public static final String LOCAL_STACK_IMAGE = "localstack/localstack:0.12.6";
+    public static final String EXASOL_DOCKER_DB_VERSION = "7.1.3";
+    public static final String LOCAL_STACK_IMAGE = "localstack/localstack:0.12.9";
```
Comment (kaklakariada) [inline src/test/java/com/exasol/cloudwatch/TestConstants.java:5]: Upgrading to latest version is not possible due to issue https://github.com/localstack/localstack/issues/5115

File: src/main/java/com/exasol/cloudwatch/configuration/EnvironmentVariableProvider.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package com.exasol.cloudwatch.configuration;
+
+/**
+ * Implementors of this interface allow access to the system's environment variables by encapsulating
+ * {@link System#getenv(String)}. The advantage over using {@link System#getenv(String)} directly is that you can mock
+ * this interface in unit tests.
+ */
+public interface EnvironmentVariableProvider {
```
Comment (jakobbraun) [inline src/main/java/com/exasol/cloudwatch/configuration/EnvironmentVariableProvider.java:8]:
```suggestion
@FunctionalInterface
public interface EnvironmentVariableProvider {
```

File: src/test/java/com/exasol/cloudwatch/configuration/EnvironmentConfigurationReaderTest.java [hunk 5]
```diff
@@ -1,59 +1,63 @@
...
 
 import java.util.List;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 class EnvironmentConfigurationReaderTest {
-    private static final EnvironmentConfigurationReader READER = new EnvironmentConfigurationReader();
     private static final List<String> AVAILABLE_METRICS = List.of("MY_METRIC");
+    private EnvironmentConfigurationReader READER;
```
Comment (jakobbraun) [inline src/test/java/com/exasol/cloudwatch/configuration/EnvironmentConfigurationReaderTest.java:14]:
```suggestion
    private EnvironmentConfigurationReader reader;
```

File: src/test/java/com/exasol/cloudwatch/configuration/EnvironmentVariableProviderTest.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+class EnvironmentVariableProviderTest {
+    @Test
+    void testMissingEnvVariableReturnsNull() {
+        assertThat(EnvironmentVariableProvider.getDefault().getEnv("this-env-variable-does-not-exist"), nullValue());
+    }
+
+    @Test
+    void testDefinedEnvVariableIsNotEmpty() {
+        assertThat(EnvironmentVariableProvider.getDefault().getEnv("PATH"), not(emptyString()));
```
Comment (jakobbraun) [inline src/test/java/com/exasol/cloudwatch/configuration/EnvironmentVariableProviderTest.java:16]: does this work on windows?

File: src/test/java/com/exasol/cloudwatch/configuration/MockEnvironmentVariableProvider.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An {@link EnvironmentVariableProvider} based on a {@link Map} that can be used in tests to simulate environment
+ * variables.
+ */
+public class MockEnvironmentVariableProvider implements EnvironmentVariableProvider {
+
```
Comment (jakobbraun) [inline src/test/java/com/exasol/cloudwatch/configuration/MockEnvironmentVariableProvider.java:11]:
```suggestion
```

File: src/test/java/com/exasol/cloudwatch/fingerprint/FingerprintExtractor.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
+package com.exasol.cloudwatch.fingerprint;
+
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+/**
+ * This class provides a method for extracting the fingerprint from an Exasol JDBC URL.
+ */
+public class FingerprintExtractor {
```
Comment (jakobbraun) [inline src/test/java/com/exasol/cloudwatch/fingerprint/FingerprintExtractor.java:9]: Why is this class in test sources?

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/cloudwatch/fingerprint/FingerprintExtractor.java (hunk 8 above)
Comment (kaklakariada) [reply src/test/java/com/exasol/cloudwatch/fingerprint/FingerprintExtractor.java:9]: We only need this for running the integration tests.

**Review by jakobbraun — APPROVED** [review]

**Pull request merged**